libs/shared/ui-components/src/SocialCard/types.ts

```diff
@@ -40,32 +46,36 @@ export interface TSocialCardProps {
    * The image will be cropped to a square on all platforms.
    * other specification's are similar to large summary cards.
    */
-  twitterImage?: string;
+  twitterImage: string;
 
   /**
    * og:image
    * absolute URL must be used rather than relative URL approach.
    * aspect ratio: The recommended size for an OG Image is 1.91:1.
    * The recommended pixel dimensions of 1200:630 px.
    */
-  ogImage?: string;
+  ogImage: string;
 
   /**
    * A text description of the image conveying the essential nature of an image to users who are visually impaired.
    * Maximum 420 characters for twitter.
    * but for Open Graph there’s no official guidance on this, but 40 characters for mobile and 60 for desktop is roughly the sweet spot.
    * More info: https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/summary-card-with-large-image
    */
-  alt?: string;
+  alt: string;
 
   /**
-   * The `summaryLargeImage` prop controls whether the large or small Twitter summary card will be used.
+   * The `twitterLargeImage` prop controls whether the large or small Twitter summary card will be used.
    * The `SocialCard` component *should* be configured with a `true` default value for this prop.
-   * Remember to use the twitter:image that fits aspect ratio of 2:1 when summaryLargeImage is true.
+   * Remember to use the twitter:image that fits aspect ratio of 2:1 when twitterLargeImage is true.
    * with minimum dimensions of 300x157 or maximum of 4096x4096 pixels (see twitterImage type above for more details).
    * full-width prominent image alongside a tweet. It is designed to give the reader a rich photo experience,
    * and clicking on the image brings the user to your website.
    * More info: `https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/summary-card-with-large-image`
    */
-  summaryLargeImage?: boolean;
-}
+  twitterLargeImage?: boolean;
```

Comment:
I think this should be an optional field of the base type.

Reply:
Ah, interesting -- and then we would need to have default images for both the large and small card formats, switchable by this field.  Agreed.

Reply:
Making it a base type means someone got to decided to use large or small twitter card even if they don't provide twitter image. I think if what is showing in the card is Quansight or labs logo we should use the large one as it's more expressive.

WDYT?

Reply:
Hm -- if `twitterLargeImage` is optional on the base type, though, that means that a blog author can omit it from the metadata -- in which case the selection would fallback to the default, and I'm fine with that default being the large size.

Reply:
but that will give the blog author a permission to use small image of Quansight or labs if they set's it `twitterLargeImage` to `false`. if you think that 's ok we need a small image for twitter.

Reply:
I say let's leave the code as is and merge it in. And if we have a real need to add support for small Twitter images for the default/fallback Quansight and Quansight Labs cards, let's add it later. 

I would prefer if we do not spend any more time shaping this component for use cases that are not actually used (yet). 

More details, if you care to read on ...

I somewhat carelessly made the suggestion to move the `twitterLargeImage` field because the first time I read 

```ts
isCustom
  ? getTwitterCardTypeName(props.twitterLargeImage)
  : getTwitterCardTypeName()
```

I thought, why not just write:

```ts
getTwitterCardTypeName(props.twitterLargeImage)
```

and then I was like, oh right, because TypeScript will complain. But I didn't think through the implication that Surafel pointed out.

For what it's worth, you could move the field to the base type and then write:

```ts
export type TSocialCardPropsDefaultImage = TSocialCardPropsBase & {
  // Our default Twitter cards for Quansight and Quansight Labs are large
  twitterLargeImage?: true;
};
```

Then you could write code in the SocialCards component that looks a bit more natural and less like a TypeScript workaround.